p2p/sql_stmt.go

```diff
@@ -48,24 +70,167 @@ func (srv *Server) initSql() error {
 		if err := srv.prepareGetRowID(); err != nil {
 			return err
 		}
-	} else {
-		log.Trace("No sql db connection info provided")
 	}
 	return nil
 }
 
+func (srv *Server) checkIfTableExists(dbName string, tableName string) (int, error) {
+	var result int
+	err := srv.DB.QueryRow(`
+		SELECT COUNT(*) 
+		FROM information_schema.TABLES 
+		WHERE (TABLE_SCHEMA = ?) AND (TABLE_NAME = ?)
+	`, dbName, tableName).Scan(&result)
+	return result, err
+}
+
+func (srv *Server) backupTable(dbName string, tableName string) error {
+	// check if table exists
+	if result, err := srv.checkIfTableExists(dbName, tableName); err != nil {
+		log.Error(fmt.Sprintf("Failed to check if %s table exists", tableName), "database", srv.MySQLName, "err", err)
+		return err
+	} else if result == 0 {
+		log.Debug(tableName+" table not found. Skipping backup", "database", srv.MySQLName)
+		return nil
+	}
+	log.Trace(tableName+" table exists", "database", srv.MySQLName)
+
+	// flush table
+	if _, err := srv.DB.Exec(fmt.Sprintf(`
+		FLUSH TABLES %s WITH READ LOCK
+	`, tableName)); err != nil {
+		log.Error(fmt.Sprintf("Failed to lock and flush %s table", tableName), "database", srv.MySQLName, "err", err)
+		return err
+	}
+	log.Trace(tableName+" table locked and flushed", "database", srv.MySQLName)
+
+	// format backup file name
+	tableNameCamel := ""
+	for _, s := range strings.Split(tableName, "_") {
+		tableNameCamel += strings.Title(s)
+	}
+	currentTime := time.Now().UTC().Format("060102-150405")
+	fileName := fmt.Sprintf("%s-%s.sql", tableNameCamel, currentTime)
+
+	// create a table backup
+	if _, err := srv.DB.Exec(fmt.Sprintf(`
+		SELECT * INTO OUTFILE '/backup/%s' FROM %s
+	`, fileName, tableName)); err != nil {
+		log.Error(fmt.Sprintf("Failed to create %s", fileName), "database", srv.MySQLName, "err", err)
+		return err
+	}
+	log.Trace(fileName+" created", "database", srv.MySQLName)
+
+	// unlock table
+	if _, err := srv.DB.Exec(`
+		UNLOCK TABLES
+	`); err != nil {
+		log.Error(fmt.Sprintf("Failed to unlock %s table", tableName), "database", srv.MySQLName, "err", err)
+		return err
+	}
+	log.Trace(tableName+" table unlocked", "database", srv.MySQLName)
+	return nil
+}
+
+func (srv *Server) dropTables() error {
+	if _, err := srv.DB.Exec(`
+		DROP TABLE IF EXISTS node_info, node_meta_info, neighbors
+	`); err != nil {
+		log.Error("Failed to drop sql tables", "database", srv.MySQLName, "err", err)
+		return err
+	}
+	log.Trace("Existing sql tables dropped", "database", srv.MySQLName)
+	return nil
+}
+
+func (srv *Server) createTables() error {
+	// create neighbors table
+	if _, err := srv.DB.Exec(`
+		CREATE TABLE IF NOT EXISTS neighbors (
```

Comment:
Do these create table statements already exist somewhere from when you initially created them? For some reason I couldn't find them in the repo

Reply:
Previously, I had python scripts to create/reset tables manually. I removed them in 23d2de5 as the node-finder now manages tables directly.
Previous workflow: run mysql server (docker container), run python scripts to generate/reset tables, run node-finder.
Current: run mysql server, run node-finder